library/src/blas3/geam_device.h

```diff
@@ -19,22 +19,30 @@ geam_device(
 
     if (tx < m && ty < n)
     {
-        if (transA == rocblas_operation_none && transB == rocblas_operation_none)
+        int a_index;
+        int b_index;
+        int c_index = tx + ldc*ty;
+
+        if (transA == rocblas_operation_none)
         {
-            C[tx + ldc*ty] = (alpha) * A[tx + lda*ty] + beta * B[tx + ldb*ty];
+            a_index = tx + ty*lda;
         }
-        else if (transA == rocblas_operation_none && transB == rocblas_operation_transpose)
+        else
         {
-            C[tx + ldc*ty] = (alpha) * A[tx + lda*ty] + beta * B[ldb*tx + ty];
+            a_index = tx*lda + ty;
         }
-        else if (transA == rocblas_operation_transpose && transB == rocblas_operation_none)
+
+        if (transB == rocblas_operation_none)
         {
-            C[tx + ldc*ty] = (alpha) * A[lda*tx + ty] + beta * B[tx + ldb*ty];
+            b_index = tx + ty*ldb;
         }
-        else if (transA == rocblas_operation_transpose && transB == rocblas_operation_transpose)
+        else
         {
-            C[tx + ldc*ty] = (alpha) * A[lda*tx + ty] + beta * B[ldb*tx + ty];
+            b_index = tx*ldb + ty;
         }
+
+        C[c_index] = alpha * A[a_index];
```

Comment:
What is the reason to break this into two statements?

Reply:
I must have misunderstood your previous comment, This was intended to address using one fma and one multiply for potentially improved accuracy.

Reply:
Thanks.  Unfortunately, the compiler isn't required to form an FMA from the expression a*b +c.  An explicit call to fma is required to ensure an fma is used.

Reply:
But will it convert to mad or mac op?

Reply:
It will in cases where it would produce the same result.

Reply:
change to use fma

Reply:
Isn't the name be gemm? Or using geam is intentional?

Reply:
Is FMA faster than MAD?

Reply:
geam is a BLAS like extension for matrix add/transpose (the transpose is out of place, not in place). It is different to gemm.

Reply:
@amcamd thanks!